source/javascripts/app/_search.js

```diff
@@ -27,24 +29,46 @@
         body: body.text()
       });
     });
+
+    determine_search_delay();
+  }
+  function determine_search_delay() {
+      if(index.tokenStore.length>5000) {
+        search_delay = 300;
+      }
   }
 
+  var wait = function() {
+    return function(executing_function, wait_time){
+          clearTimeout (timeout_handle);
+          timeout_handle = setTimeout(executing_function, wait_time);
+        };
+  }();
+
+
   function bind() {
     content = $('.content');
     searchResults = $('.search-results');
 
-    $('#input-search').on('keyup', search);
+    $('#input-search').on('keyup',function(e) {
+        wait(function(){
+          search(e);
+        }, search_delay );
```

Comment:
If we're only using `wait` in a single place, and (since it uses the global `timeout_handle`) we *only* can use it in a single place, maybe we can avoid a future person being confused about how `wait` is used by just putting its contents inline here?